R/utils.R

```diff
@@ -327,4 +335,66 @@ valid_seq <- function(params) {
   } else {
     return(FALSE)
   }
-}
\ No newline at end of file
+}
+
+resolve_prefix <- function(prefix, environment_var) {
+  if (!(is.null(prefix))) {
+    assertthat::assert_that(is.character(prefix))
+    
+    return(prefix)
+  } else {
+    prefix_env <- Sys.getenv(environment_var)
+    if (prefix_env != "") {
+      return(prefix_env)
+    } else {
+      return("/")
+    }
+  }
+}
+
+get_package_mapping <- function(script_path, url_package, dependencies) {
+  package_map <- vapply(dependencies, function(x) {
+    if (x$name %in% c('react', 'react-dom')) {
+      x$name <- 'dash_renderer'
+    }
+    dep_path <- paste(x$src$file, 
+                      x$script,
+                      sep = "/")
+  
+    # remove double slash and replace with single if present
+    result <- c(pkg_name=x$package,
+                dep_name=x$name,
+                dep_path=gsub("//", replacement = "/", dep_path)
+    )
+  }, FUN.VALUE = character(3))
+  
+  package_map <- as.data.frame(t(package_map), stringsAsFactors = FALSE)
+  
+  filename <- basename(script_path)
+  
+  pos_match <- grepl(paste0(filename, "$"), package_map[, "dep_path"]) &
```

Comment:
a comment here explaining what these variables look like would help me understand this part of the code better

Reply:
The statement generates a vector of `logical()` (Boolean) values which are used to filter the columns of the `package_map` object. 

In R, you can index a vector using a Boolean, which can be handy if the indices of elements to return are not always predictable:

```
> c(1,2,3,4)[c(FALSE, FALSE, TRUE, TRUE)]
[1] 3 4
```

The `paste0(filename, "$")` bit is used to create a regular expression requiring the filename to be the final characters in the string. Otherwise, you can run into trouble matching both `bundle.js` and `bundle.js.map` when you may only want `bundle.js`. I'll insert a succinct explanation within this codeblock.

Reply:
fixed in 8362cb5